src/toolkits/drawing_classifier/drawing_classifier.cpp

```diff
@@ -24,19 +30,122 @@ using neural_net::float_array_map;
 using neural_net::model_backend;
 using neural_net::model_spec;
 using neural_net::shared_float_array;
+using neural_net::weight_initializer;
+using neural_net::zero_weight_initializer;
+
+using padding_type = model_spec::padding_type;
+// annoymous helper sections
 
 }  // namespace
 
 std::unique_ptr<model_spec> drawing_classifier::init_model() const {
   std::unique_ptr<model_spec> result(new model_spec);
+
+  // state is updated through init_train
+  flex_string target = read_state<flex_string>("target");
+  size_t num_classes = read_state<flex_int>("num_classes");
```

Comment:
Just to confirm, this state doesn't exist yet and we will update the state in the training loop before we call `init_model` right?

Reply:
Yeah, we need to update state before init_model.